app/sync.js

```diff
@@ -325,7 +324,18 @@ module.exports.init = function (initialState) {
     if (!records || !records.length) {
       return
     }
-    syncUtil.applySyncRecords(records)
+    let showBookmarksToolbar = false
+    if (!initialState.seed) {
+      // syncing for the first time
+      const bookmarks = (AppStore.getState().get('sites') || new Immutable.List()).filter((site) => {
+        const tags = site.get('tags')
+        return tags && (tags.includes('bookmark') || tags.includes('bookmark-folder'))
+      })
+      if (!bookmarks.size) {
+        showBookmarksToolbar = true
+      }
+    }
+    syncUtil.applySyncRecords(records, showBookmarksToolbar)
```

Comment:
do you mean unnecessary with https://github.com/brave/browser-laptop/pull/7300#discussion_r101882681 or something else?